src/conversion/rustling.rs

```diff
@@ -167,33 +169,52 @@ impl OntologyFrom<Output> for SlotValue {
-            Output::TimeInterval(v) => SlotValue::TimeInterval(v.ontology_into()),
+            Output::Datetime(v) => SlotValue::InstantTime(v.ontology_into()),
+            Output::DatetimeInterval(v) => SlotValue::TimeInterval(v.ontology_into()),
         }
     }
 }
 
-pub fn convert_to_builtin(input: &str, parser_match: ParserMatch<Output>) -> BuiltinEntity {
+pub fn convert_to_builtin(input: &str, parser_match: ParserMatch<Output>, language: &Language) -> BuiltinEntity {
     BuiltinEntity {
         value: input[parser_match.byte_range.0..parser_match.byte_range.1].into(),
         range: parser_match.char_range.0..parser_match.char_range.1,
         entity: parser_match.value.clone().ontology_into(),
-        entity_kind: BuiltinEntityKind::ontology_from(&parser_match.value),
+        entity_kind: BuiltinEntityKind::ontology_from(&parser_match.value).map_to_supported(language),
     }
 }
 
 impl<'a> OntologyFrom<&'a Output> for BuiltinEntityKind {
-    fn ontology_from(v: &Output) -> Self {
-        match *v {
+    fn ontology_from(output: &Output) -> Self {
+        match *output {
             Output::AmountOfMoney(_) => BuiltinEntityKind::AmountOfMoney,
             Output::Duration(_) => BuiltinEntityKind::Duration,
             Output::Float(_) => BuiltinEntityKind::Number,
             Output::Integer(_) => BuiltinEntityKind::Number,
             Output::Ordinal(_) => BuiltinEntityKind::Ordinal,
             Output::Temperature(_) => BuiltinEntityKind::Temperature,
-            Output::Time(_) => BuiltinEntityKind::Time,
-            Output::TimeInterval(_) => BuiltinEntityKind::Time,
             Output::Percentage(_) => BuiltinEntityKind::Percentage,
+            Output::Datetime(datetime_output_value) => {
+                match datetime_output_value.datetime_kind {
+                    // Only Date and Time should occur here
+                    RustlingDatetimeKind::Date => BuiltinEntityKind::Date,
+                    RustlingDatetimeKind::Time => BuiltinEntityKind::Time,
+                    RustlingDatetimeKind::DatePeriod => BuiltinEntityKind::DatePeriod,
+                    RustlingDatetimeKind::TimePeriod => BuiltinEntityKind::TimePeriod,
+                    _ => BuiltinEntityKind::Datetime
+                }
+            },
+            Output::DatetimeInterval(datetime_interval_output_value) => {
+                match datetime_interval_output_value.datetime_kind {
+                    // Only DatePeriod and TimePeriod should occur here
+                    RustlingDatetimeKind::Date => BuiltinEntityKind::Date,
+                    RustlingDatetimeKind::Time => BuiltinEntityKind::Time,
+                    RustlingDatetimeKind::DatePeriod => BuiltinEntityKind::DatePeriod,
+                    RustlingDatetimeKind::TimePeriod => BuiltinEntityKind::TimePeriod,
+                    _ => BuiltinEntityKind::Datetime
+                }
+            },
+
         }
```

Comment:
This implementation could be simplified by using the `kind()` method of `Output`. That would give:
```rust
impl<'a> OntologyFrom<&'a Output> for BuiltinEntityKind {
    fn ontology_from(output: &Output) -> Self {
        BuiltinEntityKind::ontology_from(&output.kind())
    }
}
```